### ISSUE TITLE
Realm.create(_:value:update:) doesn't set relationships to nil when passed an Object

### ISSUE THREAD

**Issue by bdash** [realm/realm-cocoa#4926]
As mentioned in https://github.com/realm/realm-cocoa/issues/4882#issuecomment-298830486, a call to `Realm.create(_:value:update:)` that is passed an instance of an `Object` subclass with a relationship set to `nil` doesn't update an existing object's relationship to `nil`, instead leaving it set to whatever its previous value was. `Realm.add(_:update:)` does set the relationship to `nil` in this instance.

The following test case demonstrates the problem:
```swift
func testUpdateLinkWithNil() {
    let realm = try! Realm()
    var object1: SwiftLinkToPrimaryStringObject! = nil
    var object2: SwiftLinkToPrimaryStringObject! = nil
    var object3: SwiftLinkToPrimaryStringObject! = nil
    try! realm.write {
        object1 = realm.create(SwiftLinkToPrimaryStringObject.self, value: ["key1", ["key1", 1]])
        object2 = realm.create(SwiftLinkToPrimaryStringObject.self, value: ["key2", ["key2", 2]])
        object3 = realm.create(SwiftLinkToPrimaryStringObject.self, value: ["key3", ["key3", 3]])
    }
    XCTAssertNotNil(object1.object)
    XCTAssertNotNil(object2.object)
    XCTAssertNotNil(object3.object)

    try! realm.write {
        realm.create(SwiftLinkToPrimaryStringObject.self, value: ["key1", nil], update: true)
        realm.create(SwiftLinkToPrimaryStringObject.self, value: ["pk": "key2", "object": nil], update: true)
        realm.create(SwiftLinkToPrimaryStringObject.self,
                     value: SwiftLinkToPrimaryStringObject(value: ["key3", nil]), update: true)
    }

    XCTAssertNil(object1.object)
    XCTAssertNil(object2.object)
    XCTAssertNil(object3.object)
}
```

Output:
```
Test Case '-[RealmSwift_Tests.ObjectCreationTests testUpdateLinkWithNil]' started.
RealmSwift/Tests/ObjectCreationTests.swift:382: error: -[RealmSwift_Tests.ObjectCreationTests testUpdateLinkWithNil] : XCTAssertNil failed: "SwiftPrimaryStringObject {
	stringCol = key3;
	intCol = 3;
}" - 
Test Case '-[RealmSwift_Tests.ObjectCreationTests testUpdateLinkWithNil]' failed (0.121 seconds).
```

It is possible (even likely) that users are currently relying on this so we will need to fix this at a time when breaking changes are permitted.

**Comment by bdash** [comment]
It looks like @tgoyne's work in https://github.com/realm/realm-cocoa/commit/2dda22e319fb83f95acd471b076eb8661a3011de fixes this issue. We may need to find a way to preserve the existing behavior for now to avoid potentially breaking users that depend on this quirk.

**Comment by austinzheng** [comment]
Fixed by https://github.com/realm/realm-cocoa/pull/5042.

**Comment by tgoyne** [comment]
This was not fixed by #5042.

**Comment by austinzheng** [comment]
PR to fix this merged into `master-3.0`. Strange that this issue wasn't auto-closed.

**Comment by tgoyne** [comment]
Github only closes things when you merge them into the default branch.
